src/components/MobileModal/MobileModal.tsx

```diff
@@ -0,0 +1,337 @@
+/** @jsxRuntime classic /
+/** @jsx jsx */
+import { jsx, SxStyleProp } from "theme-ui"
+import React, {
+  ReactNode,
+  TouchEvent,
+  TransitionEvent,
+  useEffect,
+  useRef,
+  useState,
+} from "react"
+import ReactDOM from "react-dom"
+
+enum MobileModalTypes {
+  WITH_OVERFLOW = "overflow-herz-mobile-modal",
+  WITHOUT_OVERFLOW = "herz-mobile-modal",
+}
+
+export interface MobileModalProps {
+  backgroundStyles?: SxStyleProp
+  children?: ReactNode
+  dismissible?: (() => void) | boolean
```

Comment:
It feels a bit confusing to me to have `dismissible` potentially receive a callback or a boolean in terms of the component API.

I think separating it in `dismissible` and `onDismiss` would be more intuitive, what do you think?

Reply:
Sure, I agree. That property can be very confusing for library users.
I will fix and push into this PR.

Reply:
Fixed